src/OpenTelemetry.Instrumentation.StartupHook/StartupHook.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.IO;
+using System.Reflection;
+using System.Runtime.Loader;
+using OpenTelemetry.Instrumentation.StartupHook;
+
+/// <summary>
+/// Dotnet StartupHook
+/// </summary>
+public class StartupHook
+{
+    /// <summary>
+    /// Load and initialize OpenTelemetry.ClrProfiler.Managed assembly to bring OpenTelemetry SDK
+    /// with a pre-defined set of exporters, shims, and instrumentations.
+    /// </summary>
+    public static void Initialize()
+    {
+        AssemblyLoadContext.Default.Resolving += AssemblyResolver.LoadAssemblyFromSharedLocation;
```

Comment:
FYI this is a behavior change from the existing loader that adds an event handler to `AppDomain.CurrentDomain.AssemblyResolve`. Should this change be adopted now in this PR or added separately?

Reply:
Will get this covered in different PR, Paulo has also recommended the same.